db/migrations/00003_create_headers_table.sql

```diff
@@ -8,9 +8,26 @@ CREATE TABLE public.headers
     block_timestamp NUMERIC,
     check_count     INTEGER     NOT NULL DEFAULT 0,
     eth_node_id     INTEGER     NOT NULL REFERENCES eth_nodes (id) ON DELETE CASCADE,
+    created         TIMESTAMP   NOT NULL DEFAULT NOW(),
+    updated         TIMESTAMP   NOT NULL DEFAULT NOW(),
     UNIQUE (block_number, hash, eth_node_id)
 );
 
+-- +goose StatementBegin
+CREATE FUNCTION set_header_updated() RETURNS TRIGGER AS
+$$
+BEGIN
+    NEW.updated = NOW();
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+-- +goose StatementEnd
+
+CREATE TRIGGER header_updated
+    BEFORE UPDATE ON public.headers
+    FOR EACH ROW
```

Comment:
Probably just blatant ignorance on my side but to confirm - this only applies to rows that have actually been updated, right?

Reply:
Yep! As per [the documentation](https://www.postgresql.org/docs/9.1/sql-createtrigger.html)

> A trigger that is marked FOR EACH ROW is called once for every row that the operation modifies. For example, a DELETE that affects 10 rows will cause any ON DELETE triggers on the target relation to be called 10 separate times, once for each deleted row.